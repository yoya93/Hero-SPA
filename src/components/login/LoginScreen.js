import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);

  const handleLogin = () => {
    // history.push("/marvel");  agregar esta url en la cola

    const lastPath = localStorage.getItem("lastPath") || "/marvel";

    dispatch({
      payload: {
        name: "Yoyajr",
      },
      type: types.login,
    });
    history.replace(lastPath); //reemplaza la url en la cola
  };
  return (
    <div className="container">
      <h1>LoginScreen</h1>

      <button onClick={handleLogin} className="btn btn-primary">
        Login
      </button>
    </div>
  );
};
