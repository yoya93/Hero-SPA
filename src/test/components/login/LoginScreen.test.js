import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";

import { LoginScreen } from "../../../components/login/LoginScreen";
import { AuthContext } from "../../../auth/AuthContext";
import { types } from "../../../types/types";

describe("Pruebas en <LoginScreen />", () => {
  const contextValue = {
    dispatch: jest.fn(),
  };

  test("debe de mostrarse correctamente ", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <LoginScreen />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de realizar el dispatch y la navegacion ", () => {
    const history = {
      replace: jest.fn(),
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <LoginScreen history={history} />
      </AuthContext.Provider>
    );

    const handleClick = wrapper.find("button").prop("onClick");

    handleClick();

    expect(contextValue.dispatch).toBeCalledWith({
      payload: {
        name: "Yoyajr",
      },
      type: types.login,
    });
    expect(history.replace).toBeCalledWith("/marvel");

    localStorage.setItem("lastPath", "/dc");
    handleClick();

    expect(history.replace).toBeCalledWith("/dc");
  });
});
