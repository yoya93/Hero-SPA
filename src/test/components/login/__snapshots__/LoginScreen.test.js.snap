// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe de mostrarse correctamente  1`] = `
<div
  className="container"
>
  <h1>
    LoginScreen
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
